### DISCUSSION TITLE
DI questions

### DISCUSSION THREAD

**Discussion by KC5VDJ** [kej715/DtCyber#89] [General]
Was the scsi on the mpb-ii ever supported?  The 1991 guide says it wasn't at the time...  Why they would have it is kinda obvious, but pushing out capsule updates would be fun and thrashing (col light solid).

**Comment by KC5VDJ** [comment]
On an aside, was the trunk transceiver inductive for the esci?  I've only ever seen and messed with the vampire tap kind of 10base5.  The cables I've used even had holes in the plenum sheath at pre-measured tap points.  What about common mode in the shield?  (Need I ask after reading about the power supplies...I mean I don't tolerate more than 10-15 mV of ripple on a PC, and they are talking about funky waveforms on the building power as a result, oyfg! (And quite possibly advertising fluke multimeters without naming them directly))....

**Comment by kej715** [comment]
No, the SCSI interface was never supported. The original design intent was
that the SCSI interface would be used to support disk storage, enabling a
DI to boot itself from the disk instead of booting from a NOS or NOS/VE
host. It would also allow CDCNet to persist its service registrations on a
local disk instead of relying upon a host for storage. This would have
enabled CDCNet to be fully independent of NOS and NOS/VE, and it could have
been marketed as an independent networking solution. This timeframe
predated CISCO and other "pure" networking vendors. CDC upper management
decided against completing the SCSI feature. They decided that CDCNet
should be integral with NOS and NOS/VE; not independent of them. To the
chagrin of the CDCNet Marketing team. Huge missed opportunity!

**Reply by KC5VDJ** [reply]
That seems to be the intent of the later ve net management pc.  Abandoned for the pc?  Less expensive to upgrade.  More likely, they didn't want to toss more money at 170s.  Nahhhh...   Both.  Look at what the same bunch did to ETA...

I seem to remember that part of the 80s as a proprietary networking free-fire zone, which drove the standardization movement into mainstream.  So many casualties... 🤣🤣🤣.  But, thanks to the US Navy, some dudes at Berkeley, and ultimately Dr. Licklider having a need to go beyond telling boomers when.... That was the purpose of the eye bolt on the 1822, for a crane to lower it through the hatch.  IMP = CRM-114, all other use admin.

**Comment by KC5VDJ** [comment]
I'm wondering how realistic section 2.2.1 (baud rate) is in the bcm2835 manual.  Unfortunately it looks like tty (5-8 bit teletypes) and correspondence terminals (I would love to get the selectric! It was a work of art!) are out without baud rate conversion / buffering.  T1 looks easy, as speed goes.  CDCNet fractional t1 actually looks doable, within what the uart supports in speed, and the single-core reality.

I just had the sickest idea...a model train layout, controlled by dtcyber and pi-dis networked using t1 speed, over the lines on the poles next to the track (lmr-100, or balanced??)..  in honor of bill esrey's middle finger! 🤣🤣🤣🤣🤣. Maybe some Chas. Addams as well?

**Reply by KC5VDJ** [reply]
Ugh. Only 7 and 8 bit.  Leaves out a lot of legacy without additional circuitry.  Then again, it is a hobbyist device, where there is a will...

**Comment by kej715** [comment]
There was no “VE net management PC”. The PC associated with Cyber 180 systems such as the 930 and 2000 served as the NOS/VE system console and mainframe maintenance console (e.g., loading microcode into the mainframe at deadstart). It was connected to the mainframe via the 2-port mux. It was not connected to CDCNet.NOS and  NOS/VE provided file services to CDCNet. They ran applications that were CDCNet file and directory servers. Think of a CDCNet directory server as being analogous to a DNS server. DI’s made requests over the network to the file and directory servers hosted by NOS and NOS/VE. The channel interface between an MDI or ICA was treated as a point-to-point network link by CDCNet. It literally was the network link between an MDI/ICA and a mainframe.Cheers,Kevin On Sep 7, 2023, at 3:46 PM, KC5VDJ ***@***.***> wrote:﻿
That seems to be the intent of the later ve net management pc.  Abandoned for the pc?

**Reply by KC5VDJ** [reply]
So, the pc was like vaporware then, it is in the '91 product guide.  PC Vaporware was a thing, esp among slow moving mainframe companies at the time...

**Comment by kej715** [comment]
If you're referring to the "CDCNet Network Management Station", that was
not a PC. It was a UNIX system (either a Sun system, or a MIPS-based EP/IX
system) that ran a "port" of the CDCNet management software. It provided
the same services as NOS and NOS/VE and was introduced relatively late in
the life of CDCNet, after CDC had partnered with MIPS to co-develop UNIX
software and hardware.

**Reply by KC5VDJ** [reply]
Gotta admire the marketing of the mips consortium.  I didn't know about that.  Add CDC to the long list of companies.  Used to do a bit of SGI work (Sprint MCP team).  Interesting target systems, SGI made, Tandem logo and HA.  The result of a golf match to keep Tandem out of IMMINENT bankruptcy.  The entire financial system and telecom systems stood on the edge of collapse, all riding on a golf match arranged by a cool Irish guy, with a nice wallet and a big order, and an order to keep Tandem out of bankruptcy, probably direct from Bill Esrey's mouth, it would have been a disaster on a global scale.  I would assume this should go without explanation, at least here...they may be NonStop, but they do require support.  Graceful degradation gets pretty ugly without it.  Kinda hard to do heterogenous in the paradigm, others have tried and ⚰️⚰️⚰️⚰️⚰️... (Having worked with an etx-10 NonStop mini, and done cdr recovery from time to time on others, I do appreciate their niche, as do banks, telecom companies, the transaction based global ecosystem....).

Talk about network latency.... The MCPs (12 cpus, buku ram, oc-192 🔥burning up the lasers, mid '90s)...That speed was necessary, do you know how many phone calls happen in a day? 🤯 Tight program, probably still less than 70k lines of c, assuming 10-15k lines of growth.  New switches, new bcrs, new practices, new instructions...  was going through 802.3, terabit is now experimental...isn't erbium non-linear, thus the need for conditioners / repeaters?  There has to be a speed / distance maximum, one would think.  New coherently stimulated dopant other than erbium, (hello Corning)?  Straight glass is for LANs, maybe small MANs, and multimode; self-stimulating (pumped) single-mode is the workhorse doing long-haul (similar to the way your green laser pointer takes 1064 nm IR, and stimulates itself non-linearly, frequency doubling it to 532 nm; Nd:YAG has always been reliable stuff, iirc, the original laser was doped with it (Murray Hill)).  Up until now at least the dopant for long-haul single-mode has been erbium, where is the speed limit?

**Comment by KC5VDJ** [comment]
Where can these or a verified dump of these be found?  DI, MPB-I.  I say verified, because the windows are exposed (along with other degradation, multi-decade low-level uv (gas tubes, now leds) also causes random unpredictable erasures, from a few bits to the whole thing).  Motorola hex is acceptable, commented assembly, the holy grail.  Indeed, every eprom/prom/rom on all non eeprom boards.  (PAL/GAL, etc if possible).

![Acrobat_gejmBjc6UB_090541.png](https://github.com/kej715/DtCyber/assets/136194499/0c5f370c-798a-4136-a486-32815b9380e7)

**Reply by KC5VDJ** [reply]
Any boards with eproms will also be 68k or '010...  they will all use the same emulator for the CPU...I found an adaptable bsd-style beerware license on one that can be ripped, stripped, augmented, and adapted.  His code is competent, and would save a lot of time.  As far as 68k emulation, reinventing the wheel is wholly unnecessary.  I could, but, why?  In effect, I am going to more or less emulate a back of the magazine ad teaching board with a 68k, then the equivalent of an Atari ST or Amiga, except with a better CPU.  Zero breakage being the goal, meaning I can't be guessing at what is in the 16K of the high/low eproms on the mpb, and even moreso, the option boards.  This is more important right now than finding the capsules to drop into it from the mainframe, without these, that won't happen.

Thank Syrinx that it wasn't 2901's.

I could start with MPB-II, but I'd rather be complete, and the straight 68k would be good to get the feet wet.

Hmmm... A wifi controlled spider? Two pwm.  Four more pins, two left, two right select leg to switch L/R pwm to, the pwm controlling servos...  hmmm, fully articulating, two more lines for joint select within the leg, servos in each....to make it "economical", maybe a meter spread standing.  A daddy long legs would keep servo cost down by minimizing mass...."Lurch! Fetch my soldering iron!  Let's make Wednesday a new pet!"

**Reply by KC5VDJ** [reply]
Ps, on the 24-bit DIs, did they use the high byte of address for data?  It was a trick that was used with 68k, and MPB-I can be called memory constrained.

**Comment by KC5VDJ** [comment]
Well, the async LICs 4/8 look straightforward, z80 SIOs.

**Reply by KC5VDJ** [reply]
Hmm..  weren't dms-100 line cards using 8085's and iirc z80 sios?  Been a while.  Popular chip among telecom once it was clear CP/M was on it's deathbed.

**Comment by KC5VDJ** [comment]
This whole baud rate minimum is throwing me on some of this.

Do SIO HAT boards exist?  I would think it would be trivial.  Cyber DIs supported old and slow as well as respectably fast,  I want to hook up a Western Union if I were to feel like it, or a selectric.

**Reply by KC5VDJ** [reply]
Could always do a barrel slot multiplex and waste bandwidth......requiring external programs / hardware.

SIO HAT boards solve the problem without waste, excessive buffering, or baud rate conversion. (Or kludgy multiplex schemes just to get a slow port).

Baud rates as low as 50, 45.45, 75, 110, 150, 300....the pi itself supports as low as 400's (dependent on SYS clock).

Hell, I know people with working Western Unions, 45.45 baud 5-bit is still in service.

I am doing this for the rpi platform, but, with the exception of pi-specific stuff, it'll do any other platform.  USB serial could be another option (the platform would dictate static insertion though, I don't see an easy way to deal with dynamic insertion / removal, it can be done, but...is it worth the overhead and kludgy workarounds (vttys?), Being serial, no MAC, I have never seen one utilizing a 1-Wire or other serial number, what attaches to which? Dynamic USB would be kludgy spaghetti with little return on investment in time.  The Pi HAT idea might be the best option, short of statically connected USB serial.....

The sad thing is that the form factor is so perfect for the zeroes to fit onto a connector slapped on the back of a terminal, printer, or anything else that can be connected (with appropriate level conversion, etc).  Too bad they don't have a proper BRG that would allow for unadorned connection to low baud rate devices.

Maybe not as slow as WSJT (digital mode, very slow, used for weak signal work such as EME, where either end probably is running linear polarity (faraday rotation is in play)....soundcard dsp is fun.  Strangely, I never understood those using yagi-uda arrays...   Because of the extreme faraday rotation on the EME path (about half a million miles, two independent planetary magnetic fields), it just makes more sense to have LH/RH  pairs of Kraus Helicals in a 2x2 array.  The same drive motors / clutches / brakes would work, wind-loading could be less... Not to mention the gain per unit of size on an extremely lossy path.  Pop it on a DSN, like the 30 or so meter dishes fed by those RCA klystrons (three story building housing each of the radars on the beach formerly, a good high school friend worked in one, hope she's doing good), and Wolf 359 might be doable with WSJT.  Biggest klystrons I've ever seen IRL... I digress, but the point is made about continued need for slow async baud rates.  I still want that selectric.

I get that a PC version can get 128-256 ports per freaking slot at whatever desired rate using octopuses / multiplexers, probably more now, hell, I've used them, but I am focusing on the hobbyist side of things at the moment, along with full interfaces to each listed board in the CDCnet guide, and for rpi, an api to its functions in compass and ftn5 (pascal is good for fullscreen panels, should I do it too?).  It will generate new interest.  Maybe even teach kids speed and economy in coding, something long since flushed down the sewer by the mainstream.  It's a cyber, it's okay to get weird with it, although it could count beans (well even), the bean counters preferred IBM.  I'm sure there were stock CDC installations, but how many did semi/full custom gadgets?  This pi project carries on tradition, even adding board/board-section inter-chip protocols.  Geek out!

Tielines over i2c?  I don't see why not. A raw, reliable layer is a raw reliable layer.  If unreliable, making it error correcting, RR or whatever, it's trivial and transparent.  I won't code that, but the API will be there.  Things are fast enough for the impact of a reed-solomon code to be minimal for FEC, but always chicken/egg with this, there are voting mechanisms such as amtor/sitor offsetting that with triple character level FEC.  One out of three can be garbled.  Reed-Solomon coded streams make sense, and would have higher throughout.  My bsd background tells me the phy should be ambiguous, androgynous, and interchangeable.  If capable of a networking compatible raw stream, it is network capable.  What you do is up to you, it's a pi API, just learn cybil and 68k assembly, or use the API.

**Comment by KC5VDJ** [comment]
The PMM...  Table cache?  Noting that they didn't use the bargain basement ram chips on this board.  Still slower than bus clock...just not hideously slow like the rest.  Read/write > half a uS?  Eww.  Might as well be final-generation core, speed compatible...  even Atari and Commodore had faster ram on a 68k, and they were the princes of supply-side cheaping out (Mr. Sinclair, in the UK, being the king).  Come to think of it, the 4114, 1Kx4 @450nS, in my VIC-20 was faster than the read/write cycle in this document.  Soooooo, the bus was always thrashed with memory activity, with long wait states.  Will I have to add timing holds for the DI capsules to work?  There would probably have been early bandwidth workarounds, someone had to get weird to increase data throughout, on a unit where that is the purpose, but at face value seems crippled...then again, read the inside story from ETA..one exec daggering another?  I doubt this was five or ten way interleave... the read write times are for the board...  what is the point of the CPU running up to a magnitude of order faster than ram on a third-gen uP (4004, 8008, etc being 1st gen; the 8-bit kings from 8080 on, second; x86, 68k, 16/32k, z8k, etc being the third ), no cache....

**Reply by KC5VDJ** [reply]
Hmm..  would dec f and j uPs be 3rd or 4th gen?  11/70 compat, physically gimped tho to keep sales happy.  The 11/23 was like late 70's wasn't it?  The lsi-11 (11/03 and heathkit) was clearly second generation, workhorse tho, I class it second gen with ti's 16 bit offerings.

Interesting that the j uPs became like the ussr's "people's PC"...well, their reverse engineered versions...introducing bugs in instances (eBay is full of them).  I only recently learned about their adoration of algol-68...  Whatevuh works..

**Reply by KC5VDJ** [reply]
Why use a 16mhz CPU?  Why cheap out on ram?  These are mutually exclusive, in a non-backstabbing world.  Drama?  Reminds me of getting a fire control lock on a blackbird at age 11 (an/tpq-10)...  I sense CDC drama with this.
